src/cpuid/src/cpuid_ffi.rs

```diff
@@ -130,35 +142,33 @@ impl RawCpuid {
     fn resize(&mut self, n: u32) -> Result<(), RawCpuidResizeError> {
         // alloc
         if self.nent == 0 && n > 0 {
-            // SAFETY: `usize` will always be at least 32 bits, thus `u32` can always be safely
-            // converted into it.
-            let new_len = unsafe { usize::try_from(n).unwrap_unchecked() };
+            let new_len = from_u32(n);
 
             let new_layout =
                 Layout::array::<RawKvmCpuidEntry>(new_len).map_err(RawCpuidResizeError)?;
 
+            // JUSTIFICATION: There is no safe alternative.
             // SAFETY: Always safe.
             let new_ptr = unsafe { std::alloc::alloc(new_layout) };
```

Comment:
Why do we need what is essentially a reimplementation of `std::vec::Vec`? Can't we have a CpuId type that contains a `Vec` and From/Into for the `kvm_bindings` type, and in rust land we only use the Vec-based one?

Reply:
I think this is outside the scope of this PR.

Reply:
Somewhat disagree. If using a completely safe implementation based on `Vec` is possible, then the `JUSTIFICATION` comments should mention that, and explain why it wasn't chosen (e.g. the `This is no safe alternative` comment we have right now is wrong) - and if we cannot justify _not_ using the safe alternative, then we should use the safe alternative (which would be in-scope for this PR). If such an implementation is not possible, then there should still be some documentation on _why_ it is not possible.

Reply:
Resolving this given our discussion offline.

Reply:
Can I ask what was your conclusion? I'm also curious about this.

Reply:
We decided that replacing this will happen in a follow-up PR, and that we'll merge this one because the removal of `transmute_vec` and `unwrap_unchecked` is a change desirable on its own

Reply:
When implementing `RawCpuid` the comment is correct and there is no safe alternative. The `unsafe` usage could however be avoided by not implementing `RawCpuid` but this is a larger change that does not directly relate to the comment, this would warrant being in a separate PR if made.

Reply:
That makes sense. Thanks for your explaination!